### DISCUSSION TITLE
BUG: $connection property ignored in migrations

### DISCUSSION THREAD

**Discussion by JohnnyWalkerDigital** [laravel/framework#38783] [Q&A] [answered]
## Bug Report

- Laravel Version: 8.60.0
- PHP Version: 7.4.11
- Database Driver & Version: MySQL 8.0.25

### Description:
The [documentation](https://laravel.com/docs/8.x/migrations#setting-the-migration-connection) states that you can "set the database connection that should be used by the migration" by invoking the `$connection` property. I have done this with 70 migrations in my project and can confirm that the default connection was used for all of them. 

### Steps To Reproduce:
1. Create a second database
2. Create a new migration that you wish to use with the second database
3. Add `protected $connection = 'second_database';` to the top of your migration.
4. Run migration.

The migration should use the connection specified in the `$connection` property, but it will ignore it and use the default connection.

@driesvints I've made a video showing the issue:

[![IMAGE ALT TEXT](https://user-images.githubusercontent.com/6763222/133156831-1cbb5e96-4d98-4597-adeb-22873ce46b9c.png)](http://www.youtube.com/watch?v=CcEBEtTPmYQ "Laravel 8 - Bug")

**Comment by lk77** [comment]
I've tested it and it work :

```
class TestMigration extends Migration
{
    protected $connection = 'does_not_exist';
    [...]
}
```

```
Migrating: 2021_09_13_090345_test_migration

   InvalidArgumentException 

  Database connection [does_not_exist] not configured.

  at vendor/laravel/framework/src/Illuminate/Database/DatabaseManager.php:152
    148▕         // If the configuration doesn't exist, we'll throw an exception and bail.
    149▕         $connections = $this->app['config']['database.connections'];
    150▕ 
    151▕         if (is_null($config = Arr::get($connections, $name))) {
  ➜ 152▕             throw new InvalidArgumentException("Database connection [{$name}] not configured.");
    153▕         }
    154▕ 
    155▕         return (new ConfigurationUrlParser)
    156▕                     ->parseConfiguration($config);

      +23 vendor frames 
  24  artisan:37
      Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
```

**Reply by JohnnyWalkerDigital** [reply]
This does not follow the steps to reproduce.

**Reply by lk77** [reply]
perhaps, but it proves that laravel is trying at least to connect to that db, so it's not ignoring the database property

**Comment by lk77** [comment, marked as answer]
The $connection property does seems to be transmitted to the Illuminate/Database/Schema/Blueprint::build method


i don't know the intended behaviour, but there is no way to transmit that value to the builder currently

the only thing that make it work is to do :

```
Schema::connection($this->connection)
```
the laravel doc does not seems to state that it should work that way

https://laravel.com/docs/8.x/migrations#database-connection-table-options

the only use of $connection seems to be for the transaction

**Reply by Bassadin** [reply]
Shouldn't this be at least mentioned in the docs? I've searched for almost an hour what I did wrong there because in the docs, it just says to set that connection variable...

Your solution works, although it's not exactly pretty imho.
